# IOT Core
---

AWS IoT Core es un servicio que permite conectar de forma segura dispositivos de IoT (Internet of Things) a la nube de AWS. Facilita la captura, procesamiento y análisis de los datos generados por los dispositivos en tiempo real, utilizando protocolos como MQTT. 

AWS IoT Core gestiona millones de dispositivos y mensajes, garantizando un bajo costo por mensaje y alta disponibilidad.

In [1]:
import json

# Cargar el archivo JSON
file_path_json = 'My Estimate.json'

# Leer el archivo JSON
with open(file_path_json, encoding='latin-1') as f:
    data = json.load(f)

# Buscar el servicio MQTT (IoT Core)
for service in data['Grupos']['Servicios']:
    if service['Nombre del servicio'].strip() == 'MQTT':
        iot_service = service
        break

# Mostrar los detalles del servicio AWS IoT Core (MQTT)
print("Nombre del servicio:", iot_service['Nombre del servicio'])
print("Región:", iot_service['Región'])
print("Costo mensual:", iot_service['Costo del servicio']['mensual'])
print("Costo total (12 meses):", iot_service['Costo del servicio']['12 meses'])
print("Propiedades:")
for key, value in iot_service['Propiedades'].items():
    print(f"  {key}: {value}")


Nombre del servicio: MQTT
Región: Este de EE. UU. (Ohio)
Costo mensual: 0.18
Costo total (12 meses): 2.16
Propiedades:
  Número de dispositivos (MQTT): 50
  Número de mensajes de un dispositivo: 1
  Número medio de acciones ejecutadas por regla: 1
  Tamaño promedio de cada mensaje: 1 KB
  Tamaño promedio de cada registro: 1 KB


La configuración de AWS IoT Core (MQTT) permite conectar y gestionar hasta 50 dispositivos de manera segura. Se ha configurado para procesar mensajes con un tamaño promedio de 1 KB y un volumen bajo de acciones ejecutadas por mensaje, optimizando el costo de transmisión. 

El costo mensual estimado es de $0.18 USD.

El costo total para 12 meses es de $2.16 USD.

Este servicio permite gestionar la comunicación de los sensores de consumo energético, proporcionando una plataforma escalable y segura para procesar datos en tiempo real sin necesidad de infraestructura física compleja. La configuración actual asegura que los costos sean mínimos, pero que la plataforma pueda escalar para manejar más dispositivos en el futuro.